### PR TITLE
New sample: Configure subnetwork trace

### PR DESCRIPTION
**Pull request by ZackAllen**

### CODE REVIEW

File: src/Forms/Shared/Samples/Network analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.xaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                        </Grid>
+                    </ViewCell>
+                </TableSection>
+                <TableSection Title="Barrier conditions">
+                    <ViewCell>
+                        <ScrollView HeightRequest="150">
+                            <Label
+                                x:Name="ConditionBarrierExpression"
+                                Grid.Row="0"
+                                FontSize="Medium"
```
Comment (nCastle1) [inline src/Forms/Shared/Samples/Network analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.xaml]: I think setting a Bold font might be better than font size here. The large fonts feel a bit jarring.

File: src/Forms/UWP/App.xaml [hunk 2]
```diff
@@ -4,5 +4,13 @@
     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
     xmlns:local="using:ArcGISRuntime.UWP"
     RequestedTheme="Light">
-
+    <Application.Resources>
+        <DataTemplate x:Key="TableSection">
+            <TextBlock
+                Margin="0,20,0,0"
```
[the remark this answers is not in the document]
Comment (ZackAllen) [reply src/Forms/UWP/App.xaml]: It is for working around a Forms.UWP bug where headers for table sections are made to be all lowercase - https://forums.xamarin.com/discussion/64281/uwp-prevent-all-lowercase-table-section-headers

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Network analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        {
+            try
+            {
+                Configuration.IsEnabled = false;
+
+                // Create and load the utility network.
+                _utilityNetwork = await UtilityNetwork.CreateAsync(new Uri(FeatureServiceUrl));
+
+                // Build the choice lists for network attribute comparison.
+                Attributes.ItemsSource = _utilityNetwork.Definition.NetworkAttributes.Where((i => i.IsSystemDefined == false));
```
Comment (nCastle1) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Network analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.xaml.cs]: I think this Where clause can be simplified to `i => !i.IsSystemDefined`. Also note that there are extra enclosing parentheses. 

I would recommend trying to make the variable names a little clearer in the lambdas. Not a huge deal, but something like `.Where(netattr => !netAttr.IsSystemDefined)` would make the intent clearer.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Network analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+using System.Windows.Controls;
+
+namespace ArcGISRuntime.WPF.Samples.ConfigureSubnetworkTrace
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Configure subnetwork trace",
+        "Network analysis",
+        "Get a server-defined trace configuration for a given tier and modify its traversability scope, add new condition barriers and control what is included in the subnetwork trace result.",
+        "")]
+    [ArcGISRuntime.Samples.Shared.Attributes.OfflineData()]
```
Comment (nCastle1) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Network analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.xaml.cs]: I recommend removing the unneeded offline data attribute

**Pull request merged**